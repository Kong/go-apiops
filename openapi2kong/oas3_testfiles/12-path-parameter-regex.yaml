# The path parameters should be converted to a regex for matching the route.
# Including creating captures of the parameter values.

openapi: 3.0.3
info:
  title: Path parameter test
  version: v1
servers:
  - url: "https://example.com"

paths:
  # simple regex capture names. Also the path has a trailing slash.
  /demo/{something}/else/{todo}/:
    get:
      operationId: opsid
      parameters:
        - in: path
          name: something
          required: true
          schema:
            type: string
        - in: path
          name: todo
          required: true
          schema:
            type: string
      responses:
        "200":
          description: OK
  # note that the names are not valid regex capture names, hence they must be sanitized
  # in the output. Also the path has NO trailing slash.
  /demo/{ white space }/{x*()_-+=x}/{-To_Do}:
    get:
      operationId: opsid
      parameters:
        - in: path
          name: " white space "
          required: true
          schema:
            type: string
        - in: path
          name: "x*()_-+=x"
          required: true
          schema:
            type: string
        - in: path
          name: -To_Do
          required: true
          schema:
            type: string
      responses:
        "200":
          description: OK
  # some more complex paths, with and without parameters
  # examples from: https://github.com/Kong/insomnia/pull/5246
  /batchs(Material='iron',Batch='10'):
    get:
      responses:
        "200":
          description: no parameters
      operationId: getBatchNoParams
  /batchs(Material='{Material}',Batch='{Batch}'):
    get:
      parameters:
        - in: path
          name: Material
          required: true
          schema:
            type: string
        - in: path
          name: Batch
          required: true
          schema:
            type: string
      responses:
        "200":
          description: An echo message.
      operationId: getBatchWithParams
    post:
      # test validating not escaping the path twice; we have 2 methods on 1 path
      parameters:
        - in: path
          name: Material
          required: true
          schema:
            type: string
        - in: path
          name: Batch
          required: true
          schema:
            type: string
      responses:
        "200":
          description: An echo message.
      operationId: postBatchWithParams
  # dollar sign in path
  /some/$test:
    get:
      responses:
        "200":
          description: no parameters
      operationId: dollar-sign
