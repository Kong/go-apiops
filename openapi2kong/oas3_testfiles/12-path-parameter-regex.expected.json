{
  "_format_version": "3.0",
  "services": [
    {
      "host": "example.com",
      "id": "3e0fefa1-06b5-589a-ab2d-2d01c70bf55a",
      "name": "Path_parameter_test",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "d24ac8a3-d744-5952-89a9-79cffecbf8dd",
          "methods": [
            "GET"
          ],
          "name": "Path_parameter_test-getBatchNoParams",
          "paths": [
            "~/batchs\\(Material='iron',Batch='10'\\)$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        },
        {
          "id": "928231e6-951a-50ff-8891-bc9f9d12538b",
          "methods": [
            "GET"
          ],
          "name": "Path_parameter_test-getBatchWithParams",
          "paths": [
            "~/batchs\\(Material='(?<material>[^#?/]+)',Batch='(?<batch>[^#?/]+)'\\)$"
          ],
          "plugins": [],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        },
        {
          "id": "95358c02-e755-5a0c-9044-bf9a9b4e1561",
          "methods": [
            "POST"
          ],
          "name": "Path_parameter_test-postBatchWithParams",
          "paths": [
            "~/batchs\\(Material='(?<material>[^#?/]+)',Batch='(?<batch>[^#?/]+)'\\)$"
          ],
          "plugins": [],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        },
        {
          "id": "2f9299ba-2694-52a8-8e82-bd2d3446bc41",
          "methods": [
            "GET"
          ],
          "name": "Path_parameter_test-opsid",
          "paths": [
            "~/demo/(?<something>[^#?/]+)/else/(?<to_do>[^#?/]+)/$"
          ],
          "plugins": [],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_12-path-parameter-regex.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_12-path-parameter-regex.yaml"
      ]
    }
  ],
  "upstreams": []
}
